vyper/builtin_functions/functions.py

```diff
@@ -2576,6 +2568,29 @@ def _eval_decimal(self, type_):
         return typinfo.decimal_bounds[1]
 
 
+class Epsilon(FoldedFunction):
+
+    _inputs = [("typename", "TYPE_DEFINITION")]
+    _id = "epsilon"
+
+    def evaluate(self, node):
+        self._validate_arg_types(node)
+        input_type = get_type_from_annotation(node.args[0], DataLocation.UNSET)
+
+        if not isinstance(input_type, FixedAbstractType):
+            raise InvalidType(f"Expected decimal type but got {input_type} instead", node)
+
+        # this check seems redundant, but sets a pattern to be followed
+        # when new decimal types are created
+        if isinstance(input_type, DecimalDefinition):
+            val = self._eval_decimal(input_type)
```

Comment:
i think you can just inline `_eval_decimal` here. the use of that pattern elsewhere is to deal with some kind of inheritance, not needed here.